sockjs/websocket.go

```diff
@@ -1,23 +1,32 @@
 package sockjs
 
 import (
-	"fmt"
 	"net/http"
-	"strings"
 
 	"github.com/gorilla/websocket"
 )
 
 // WebSocketReadBufSize is a parameter that is used for WebSocket Upgrader.
 // https://github.com/gorilla/websocket/blob/master/server.go#L230
+//
+// Deprecated: Set WebSocketUpgrader.ReadBufferSize instead.
 var WebSocketReadBufSize = 4096
 
 // WebSocketWriteBufSize is a parameter that is used for WebSocket Upgrader
 // https://github.com/gorilla/websocket/blob/master/server.go#L230
+//
+// Deprecated: Set WebSocketUpgrader.WriteBufferSize instead.
 var WebSocketWriteBufSize = 4096
 
+// WebSocketUpgrader is used to configure websocket handshakes and websocket
+// connection details.
+var WebSocketUpgrader = &websocket.Upgrader{
+	ReadBufferSize:  WebSocketReadBufSize,
+	WriteBufferSize: WebSocketWriteBufSize,
+}
+
 func (h *handler) sockjsWebsocket(rw http.ResponseWriter, req *http.Request) {
-	conn, err := websocket.Upgrade(rw, req, nil, WebSocketReadBufSize, WebSocketWriteBufSize)
+	conn, err := WebSocketUpgrader.Upgrade(rw, req, nil)
```

Comment:
Good points, I've added values for Error and CheckOrigin fields and set buffers to 0 to reuse the http server's ones.

If we're ok to make backward-incompatible change - ignoring WebSocket{Read,Write}BufSize in favour of WebSocketUpgrader, then we should also be ok with changing the defaults for the buffers.